### ISSUE TITLE
Compilation error with embedded domains and postgres types

### ISSUE THREAD

**Issue by berlam** [jOOQ/jOOQ#14916]
### Expected behavior

Generating an embedded domain of a postgres type should create compilable code when using the postgres-extension.

```sql
CREATE DOMAIN abc int4range
```

### Actual behavior

The generated code is not compilable because the data type does not match the `IntegerRange` return type.

```java
public class Domains {

    public static final Domain<IntegerRange> ABC = Internal.createDomain(
          schema()
        , DSL.name("abc")
        , org.jooq.impl.DefaultDataType.getDefaultDataType("\"public\".\"int4range\"")
    );

    private static final Schema schema() {
        return new LazySchema(DSL.name("public"), DSL.comment(""), () -> Public.PUBLIC);
    }
}
```

### Steps to reproduce the problem

                <configuration>
                    <generator>
                        <database>
                            <name>org.jooq.meta.postgres.PostgresDatabase</name>
                            <includes>.*</includes>
                            <excludes>flyway_schema_history</excludes>
                            <inputSchema>public</inputSchema>
                            <schemaVersionProvider>SELECT :schema_name || '_' || MAX("version") FROM "flyway_schema_history"</schemaVersionProvider>
                            <recordVersionFields>version</recordVersionFields>
                            <recordTimestampFields>modified_at</recordTimestampFields>
                            <forceIntegerTypesOnZeroScaleDecimals>true</forceIntegerTypesOnZeroScaleDecimals>
                            <embeddableDomains>.*</embeddableDomains>
                        </database>
                        <generate>
                            <jpaAnnotations>false</jpaAnnotations>
                            <generatedAnnotation>true</generatedAnnotation>
                            <generatedAnnotationType>DETECT_FROM_JDK</generatedAnnotationType>
                            <generatedAnnotationDate>false</generatedAnnotationDate>
                            <nullableAnnotation>true</nullableAnnotation>
                            <nullableAnnotationType>javax.annotation.Nullable</nullableAnnotationType>
                            <nonnullAnnotation>true</nonnullAnnotation>
                            <nonnullAnnotationType>javax.annotation.Nonnull</nonnullAnnotationType>
                            <constructorPropertiesAnnotation>true</constructorPropertiesAnnotation>
                            <javaTimeTypes>true</javaTimeTypes>
                            <intervalTypes>true</intervalTypes>
                            <jsonTypes>true</jsonTypes>
                            <generatedSerialVersionUID>HASH</generatedSerialVersionUID>
                        </generate>
                    </generator>
                </configuration>

### jOOQ Version

jOOQ Professional 3.18.2

### Database product and version

PostgresSQL 13.7

### Java Version

openjdk 11.0.18 2023-01-17

### OS Version

Debian Sid

### JDBC driver name and version (include name if unofficial driver)

org.postgresql:postgresql:42.5.1

**Comment by lukaseder** [comment]
Thanks for your report. I see this is an oversight from https://github.com/jOOQ/jOOQ/issues/5507. The workaround might be to add explicit converters to the domain type?

**Comment by lukaseder** [comment]
No, that workaround doesn't work. The `JavaGenerator` doesn't generate any converters or bindings on a `DOMAIN` type. Also, the referencing table cannot resolve the type if a `DOMAIN` is used in a table.

**Comment by lukaseder** [comment]
Fixed in jOOQ 3.19.0, 3.18.3 (https://github.com/jOOQ/jOOQ/issues/14920), 3.17.12 (https://github.com/jOOQ/jOOQ/issues/14921), and 3.16.18 (https://github.com/jOOQ/jOOQ/issues/14922).

Thanks again for your report!

**Comment by lukaseder** [comment]
The backports may be available in snapshot builds only by mid next week.
